libvast/vast/defaults.hpp

```diff
@@ -149,6 +149,25 @@ struct pcap {
 
 } // namespace import
 
+// -- constants for the explore command and its subcommands --------------------
+
+namespace explore {
+
+// A value of zero means 'unlimited' for all three limits below.
+// If all limits are non-zero, the number of results is bounded
+// by `min(max_events, limit1*limit2)`.
+
+/// Maximum total number of results.
+constexpr size_t max_events = 0;
+
+/// Maximum number of results for the initial query.
+constexpr size_t limit1 = 1000;
+
+/// Maximum number of results for every explored context.
+constexpr size_t limit2 = 1000;
```

Comment:
Do you think there'll ever be more than two stages? in that case, would a `vector<size_t>` make more sense? We could name is `max_events_per_stage`.